docs/03_tutorials/20_tic-tac-toe-game-smart-contract-single node.md

```diff
@@ -0,0 +1,761 @@
+
+For example, in the above game board illustration 
+
+* An empty board is represented as [0, 0, 0, 0, 0, 0, 0, 0, 0].
+* The host takes the first move and marks their square with an X in row 0, column 2, which results in an array represented as [0, 0, 1, 0, 0, 0, 0, 0, 0].
+* The challenger takes the next move and marks their square with an O in row 0, column 1 which results in an array represented as [0, 2, 1, 0,  0, 0, 0, 0, 0].
+
+At the end of a drawn game, the board array may be  represented as [1, 2, 1, 1, 2, 2, 2, 1, 1] 
+
+Array Representation
+
+| Row       | 0     | 1     | 2     |
+| :-------: | :---: | :---: | :---: |
+|     0     |   1   |   2   |   1   |
+|     1     |   1   |   2   |   2   |
+|     2     |   2   |   1   |   1   |
+
+Game Representation
+
+| Row       | 0     | 1     | 2     |
+| :-------: | :---: | :---: | :---: |
+|     0     |   x   |   o   |   x   |
+|     1     |   x   |   o   |   o   |
+|     2     |   o   |   x   |   x   |
+
+### Create tictactoe.hpp file
+This section creates the tictactoe.hpp file. This header file contains the declarations of the tictactoe class, the definitions of tictactoe game data structures, and the declarations of tictactoe game methods, known as actions in EOSIO smart contracts.
+
+
+#### Game Data Structures
+The tic-tac-toe smart contract hpp file defines the following public data structures to store game information.  
+
+* game - The game data structure contains game data. The structure has variables of type eosio::name, for challenger, host, turn and winner. Click on this link for a definition of [eosio::name](https://developers.eos.io/manuals/eosio.cdt/v1.7/structeosio_1_1name "eosio.cdt reference - name structure") . The structure has a vector representing the game board  
+* Games - Games is a type definition that uses a class template. Games uses an eosio::muti_index template to define a type which stores a game structure in RAM. Click on this link for more information on [eosio::multi_index](https://developers.eos.io/manuals/eosio.cdt/v1.7/group__multiindex "eosio.cdt reference - multi index table") and click on this link for more general information about [multi index tables](https://developers.eos.io/welcome/latest/getting-started/smart-contract-development/data-persistence) 
+
+#### Game Actions
+The tic-tac-toe smart contract hpp file defines the following four public actions to operate the game logic.  
+
+* Create - This action launches a new game and creates a new game board array. The host may use this command.
+* Restart - This action clears data from an existing game board array. The host or the challenger may use this command.
+* Close - This action deletes and removes existing game data and frees up any storage the game uses. No game data persists. The host may use this command.
+* Move - This action sets a marker on the gameboard and updates the game board array. The host or the challenger may use this command.
```

Comment:
Smart contract actions should be lowercase to avoid confusion. I'd also suggest using single codeblocks for these:
"Create" --> "\`create\`"
"Restart" --> "\`restart\`"
"Close" --> "\`close\`"
"Move" --> "\`move\`"